Tensile/cmake/TensileConfig.cmake

```diff
@@ -184,7 +187,8 @@ function(TensileCreateLibraryFiles
   endif()
 
   if(Tensile_ARCHITECTURE)
-    set(Options ${Options} "--architecture=${Tensile_ARCHITECTURE}")
+    string (REPLACE ";" "\\\\\\\;" archString "${Tensile_ARCHITECTURE}")
```

Comment:
each time we assign the CMake list to another list, it eats one of the escapes, and when the escapes run out it interprets the semicolon as a list delimiter as that's how CMake handles lists. This would then separate the archs into individual options, which breaks the call. This is just enough escapes to make it to the final call into the process while still maintaining the semicolons between the archs. Would welcome a better alternative.

Reply:
If kept inside quotes it should remain a string with semicolons.  If used outside the quotes then cmake will see it as a list, or if used with a list. It may be you can just add quotes around later uses like "${Options}" and anything like the command it gets embedded inside then also needs to be quoted, not much better, but you don't have to count how may future times it is copied, and some are inside an if statement?

Reply:
updated